fenix/app/src/main/java/org/mozilla/fenix/translations/preferences/downloadlanguages/DownloadLanguageFileDialogPreference.kt

```diff
@@ -0,0 +1,195 @@
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Firefox Preference: Download Languages Delete Dialog compose view.
+ *
+ * @param fileSize Language file size in bytes that should be displayed in the dialogue title.
+ * @param isCheckBoxEnabled Whether saving mode checkbox is checked or unchecked.
+ * @param isAllLanguagesItemType Whether the download language file item is of type all languages.
+ * @param onSavingModeStateChange Invoked when the user clicks on the checkbox of the saving mode state.
+ * @param onConfirmDownload Invoked when the user click on the "Download" dialog button.
+ * @param onCancel Invoked when the user clicks on the "Cancel" dialog button.
+ */
+@Suppress("LongParameterList")
+@Composable
+fun DownloadLanguageFileDialogPreference(
+    fileSize: Long,
+    isCheckBoxEnabled: Boolean,
+    isAllLanguagesItemType: Boolean,
+    onSavingModeStateChange: (Boolean) -> Unit,
+    onConfirmDownload: () -> Unit,
+    onCancel: () -> Unit,
+) {
+    Dialog(onDismissRequest = {}) {
+        Column(
+            modifier = Modifier
+                .background(
+                    color = FirefoxTheme.colors.layer2,
+                    shape = RoundedCornerShape(8.dp),
+                )
+                .padding(top = 16.dp, bottom = 16.dp),
+        ) {
+            Text(
+                text = stringResource(
+                    R.string.download_language_file_dialog_title,
+                    fileSize.toMegabyteOrKilobyteString(),
+                ),
+                modifier = Modifier
+                    .semantics { heading() }
+                    .padding(start = 16.dp, end = 16.dp),
+                color = FirefoxTheme.colors.textPrimary,
+                style = FirefoxTheme.typography.headline7,
+                fontWeight = FontWeight.Bold,
+            )
+
+            if (isAllLanguagesItemType) {
+                Text(
+                    text = stringResource(
+                        R.string.download_language_file_dialog_message_all_languages,
+                    ),
+                    modifier = Modifier.padding(16.dp),
+                    style = FirefoxTheme.typography.body2,
+                    color = FirefoxTheme.colors.textPrimary,
+                )
+            }
+
+            DownloadLanguageFileDialogCheckbox(
+                isCheckBoxEnabled = isCheckBoxEnabled,
+                onSavingModeStateChange = onSavingModeStateChange,
+            )
+
+            val primaryButtonText: String = if (isAllLanguagesItemType) {
+                stringResource(id = R.string.download_language_file_dialog_positive_button_text_all_languages)
+            } else {
+                stringResource(id = R.string.download_language_file_dialog_positive_button_text)
+            }
+
+            PrimaryButton(
+                text = primaryButtonText,
+                modifier = Modifier
+                    .padding(start = 16.dp, end = 16.dp, top = 16.dp)
```

Comment:
Can these padding be implemented in a way using `Spacer()` or by adding it to the above `Column` since we are repeating the horizontal padding in just about every element?

Reply:
Done